docs/polkadot/networking/3-attestation.md

```diff
@@ -1,54 +1,268 @@
 
-### Direct Routing: sending/ receiving erasure coded pieces
+Let T_b be T_B divided by the number of validators per parachain, i.e. T_b = T_B/(N/C). This represents the average time it takes for a distributor to transfer the pieces of a block if this responsibility was shared equally across all validators of a parachain.
+
+Let T_L represent the standard deviation of the distribution of when validators begin this distribution process for each period.
+
+In practise based on some crude real-world measurements, we guess that T_l is about 150ms, T_b is about 3s, and T_L is about 750ms. In other words, T_b ~= 20 * T_l ~= 4 * T_L.
+
+## High-level requirements
+
+Ensure that ceil(N/3) of the pieces of all C blocks remains available and retrievable for a reasonable amount of time, across the N validators.
+
+## Design considerations
+
+### General aims
+
+A1: Reduced redundancy & latency: pieces should (mostly) only be sent to the nodes that need it, i.e. its main storer, and the secondary checkers.
+
+A2: Constrain the use of shared OS resources, so the protocol does not interfere with other programs running at the same site.
+
+A3: Performance degrades reasonably under load or imperfect network conditions.
+
+A4: TODO: Protocols should not be easily spammable, or the spammers should be easily identifiable and punishable.
+
+### Special considerations
+
+S1: A small fraction of nodes may have poor reachability, and need to communicate via a proxy. Otherwise, we can assume the graph of communication links is mostly-connected via the public internet.
+
+This is an optional consideration; we believe we do not need to consider it in the initial implementation, and so our solution for this is described in a separate section in this document. Parts of the main proposal that will need to be changed or extended for this, are marked [TBX S1 #($ref)], TBX standing for "to be extended".
+
+## Protocol overview
+
+We rely on an underlying gossip network that allows us to broadcast various metadata to everyone, namely:
+
+1. receipts of specific pieces ("I have X piece")
+2. receipts of "enough" pieces ("I have >1/3 of pieces, I don't need more"). This could also double as an attestation to everyone else, that there is availability from the attester's position on the network.
+3. [TBX S1 #1]
+
+This metadata should be gossiped every few seconds. The data of the actual pieces are distributed via a separate topology as described below:
+
+Recall that we have a disjoint partition of N validators into C sets of parachain validators, each set having size N/C. For our purposes for this subprotocol, we will randomly assign a co-ordinate (c, i) to every validator, with c in [0, C) and i in [0, N/C). Fixing c and varying i defines a particular parachain validator set; varying c and fixing i defines what we'll call a particular validator "ring". This name is only meant to be very slightly suggestive, the precise structure and its justification will be described below.
+
```

Comment:
```suggestion

Example:

Let's say we have 20 validators `[a, b, c, ..., t]` and 5 parachains. The co-ordinates of the validators could look like such `[a: (0, 0), b: (0, 1), c: (0, 2), d: (0,3), e: (1, 0), ..., t: (3, 3)]`. The validator set of the first parachain would be `[a, b, c, d]`. The first validator ring would be `[a, e, i, m, q]`, the second `[b, f, j, n, r]`.

```

Reply:
Thanks for the example, I made a minor fix - the co-ord for `t` should be `(4, 3)`.